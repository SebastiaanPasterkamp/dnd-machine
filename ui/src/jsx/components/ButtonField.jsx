import React from 'react';
import PropTypes from 'prop-types';

import utils from '../utils.jsx';

import LazyComponent from './LazyComponent.jsx';

export class ButtonField extends LazyComponent
{
    render() {
        const {
            color, icon, className, label, ...buttonProps
        } = this.props
        const style = utils.makeStyle({
            [color]: color,
            'icon': icon,
            ['fa-' + icon]: icon
        }, ['nice-btn', className]);

        return <button
                {...buttonProps}
                className={style}
                >
            {label}
        </button>
    }
}

ButtonField.defaultProps = {
    label: "Button",
    onClick: () => {
        console.log(['ButtonField']);
    }
};

ButtonField.propTypes = {
    label: PropTypes.string,
    color: PropTypes.string,
    icon: PropTypes.string,
    className: PropTypes.string
};

export default ButtonField;
