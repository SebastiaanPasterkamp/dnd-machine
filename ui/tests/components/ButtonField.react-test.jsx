import React from 'react';
import ButtonField from '../../src/jsx/components/ButtonField.jsx';
import renderer from 'react-test-renderer';

describe('Component: ButtonField', () => {
    it('should show a simple button', () => {
        const tree = renderer.create(
            <ButtonField />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should show an icon button', () => {
        const tree = renderer.create(
            <ButtonField
                icon="thumbs-up"
                label="Okay"
                />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should show a colored button', () => {
        const tree = renderer.create(
            <ButtonField
                color="good"
                label="Good"
                />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should show a custom button', () => {
        const tree = renderer.create(
            <ButtonField
                className="special"
                />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
