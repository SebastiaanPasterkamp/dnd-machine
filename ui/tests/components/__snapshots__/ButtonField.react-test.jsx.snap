// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ButtonField should show a colored button 1`] = `
<button
  className="nice-btn good"
  onClick={[Function]}
>
  Good
</button>
`;

exports[`Component: ButtonField should show a custom button 1`] = `
<button
  className="nice-btn special"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Component: ButtonField should show a simple button 1`] = `
<button
  className="nice-btn"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Component: ButtonField should show an icon button 1`] = `
<button
  className="nice-btn icon fa-thumbs-up"
  onClick={[Function]}
>
  Okay
</button>
`;
